test/fake_http_server.cc

```diff
@@ -43,20 +51,104 @@ void FakeServer::SetResponse(const std::string& path,
   handler_.path_responses[path] = response;
 }
 
+void FakeServer::AllowStream(const std::string& path) {
+  handler_.path_streams[path] = std::unique_ptr<Handler::Stream>(
+      new Handler::Stream);
+}
+
+bool FakeServer::WaitForOneStreamWatcher(const std::string& path) {
+  auto stream_it = handler_.path_streams.find(path);
+  if (stream_it == handler_.path_streams.end()) {
+    LOG(ERROR) << "No stream for path " << path;
+    return false;
+  }
+  auto& stream = stream_it->second;
+  std::unique_lock<std::mutex> queues_lock(stream->mutex);
+  bool success = stream->cv.wait_for(queues_lock,
+                                     std::chrono::seconds(3),
+                                     [&stream]{return stream->queues.size() > 0;});
+  queues_lock.unlock();
```

Comment:
This happens automatically in the destructor. So the code can just be:
```c++
  auto& stream = stream_it->second;
  std::unique_lock<std::mutex> queues_lock(stream->mutex);
  return stream->cv.wait_for(queues_lock,
                             time::seconds(3),
                             [&stream]{ return stream->queues.size() > 0; });
```

Reply:
Done.